fearless/CoreLayer/OperationFactory/BlockExplorer/History/Main/HistoryOperationFactory.swift

```diff
@@ -8,19 +8,27 @@ final class HistoryOperationFactoriesAssembly {
     static func createOperationFactory(
         chainAsset: ChainAsset,
         txStorage: AnyDataProviderRepository<TransactionHistoryItem>,
-        runtimeService: RuntimeCodingServiceProtocol
-    ) -> HistoryOperationFactoryProtocol {
+        runtimeService: RuntimeCodingServiceProtocol?
+    ) -> HistoryOperationFactoryProtocol? {
         switch chainAsset.chain.externalApi?.history?.type {
         case .subquery:
+            guard let runtimeService = runtimeService else {
+                return nil
+            }
+
             return SubqueryHistoryOperationFactory(txStorage: txStorage, runtimeService: runtimeService)
         case .subsquid:
-            return SubsquidHistoryOperationFactory(txStorage: txStorage, runtimeService: runtimeService)
+            return SubsquidHistoryOperationFactory(txStorage: txStorage)
         case .giantsquid:
-            return GiantsquidHistoryOperationFactory(txStorage: txStorage, runtimeService: runtimeService)
+            return GiantsquidHistoryOperationFactory(txStorage: txStorage)
         case .sora:
             return SoraHistoryOperationFactory(txStorage: AnyDataProviderRepository(txStorage))
+        case .alchemy:
+            return AlchemyHistoryOperationFactory(txStorage: txStorage, alchemyService: AlchemyService())
+        case .etherscan:
+            return EtherscanHistoryOperationFactory()
         case .none:
-            return GiantsquidHistoryOperationFactory(txStorage: txStorage, runtimeService: runtimeService)
+            return GiantsquidHistoryOperationFactory(txStorage: txStorage)
```

Comment:
why with none case we return GiantsquidHistoryOperationFactory? Perhaps should return nil?